src/connectors/breadcrumb/connectBreadcrumb.js

```diff
@@ -0,0 +1,194 @@
+  */
+export default function connectBreadcrumb(renderFn) {
+  checkRendering(renderFn, usage);
+  return (widgetParams = {}) => {
+    const { attributes, separator = ' > ', rootPath = null } = widgetParams;
+    const [hierarchicalFacetName] = attributes;
+
+    return {
+      getConfiguration: currentConfiguration => {
+        if (currentConfiguration.hierarchicalFacets) {
+          const facetSet = find(
+            currentConfiguration.hierarchicalFacets,
+            ({ name }) => name === hierarchicalFacetName
+          );
+          if (facetSet) {
+            if (
+              !isEqual(facetSet.attributes, attributes) ||
+              facetSet.separator !== separator
+            ) {
+              console.warn(
+                'Using Breadcrumb & HierarchicalMenu on the same facet with different options'
+              );
+            }
+            return {};
+          }
+        }
+
+        return {
+          hierarchicalFacets: [
+            {
+              attributes,
+              name: hierarchicalFacetName,
+              separator,
+              rootPath,
+            },
+          ],
+        };
+      },
+
+      init({ createURL, helper, instantSearchInstance }) {
+        this._createURL = facetValue => {
+          if (!facetValue) {
+            const breadcrumb = helper.getHierarchicalFacetBreadcrumb(
+              hierarchicalFacetName
+            );
+            if (breadcrumb.length > 0) {
+              return createURL(
+                helper.state.toggleRefinement(
+                  hierarchicalFacetName,
+                  breadcrumb[0]
+                )
+              );
+            } else {
+              return undefined;
+            }
+          }
+          return createURL(
+            helper.state.toggleRefinement(hierarchicalFacetName, facetValue)
+          );
+        };
+
+        this._refine = function(facetValue) {
+          if (!facetValue) {
+            const breadcrumb = helper.getHierarchicalFacetBreadcrumb(
+              hierarchicalFacetName
+            );
+            if (breadcrumb.length > 0) {
+              helper
+                .toggleRefinement(hierarchicalFacetName, breadcrumb[0])
+                .search();
+            }
+          } else {
+            helper.toggleRefinement(hierarchicalFacetName, facetValue).search();
+          }
+        };
+
+        renderFn(
+          {
+            createURL: this._createURL,
+            canRefine: false,
+            instantSearchInstance,
+            items: [],
+            refine: this._refine,
+            widgetParams,
+          },
+          true
+        );
+      },
+
+      render({ instantSearchInstance, results, state }) {
+        if (
+          !state.hierarchicalFacets ||
+          (Array.isArray(state.hierarchicalFacets) &&
+            state.hierarchicalFacets.length === 0)
+        ) {
+          throw new Error(usage);
```

Comment:
This message is sent too late. It should be done as soon as we have all the necessary information. Most likely before returning the widget object.